ogr/ogrsf_frmts/ogrsf_frmts.h

```diff
@@ -395,6 +401,33 @@ class CPL_DLL OGRLayer : public GDALMajorObject
 
   protected:
     //! @cond Doxygen_Suppress
+
+    enum class FieldChangeType : char
+    {
+        ADD,
+        ALTER,
+        DELETE
```

Comment:
It appears there must be a #define DELETE somewhere in a Windows system header...  So maybe switch to Add/Alter/Delete or add a _FIELD suffix ?